### PR TITLE
Remove extra prerun frame, Add dynamic time difference

### PR DESCRIPTION
**Pull request by deadw1nter**
Thanks Kamay & ici & Promises & Nairda for help.

# Changes
- Fix an issue that prerun frame size is bigger than intended size when ``shavit_replay_prerun_always`` is 0
- Add dynamic time difference (#818)

# Something I can't do
- [ ] Polish native name and description

### CODE REVIEW

File: addons/sourcemod/scripting/shavit-replay.sp [hunk 1]
```diff
@@ -1588,17 +1606,40 @@ public void DeleteFrames(int client)
 	delete gA_PlayerFrames[client];
 }
 
+int g_iLoopFrame[MAXPLAYERS+1];
```
Comment (kidfearless) [inline addons/sourcemod/scripting/shavit-replay.sp:1609]: delete

File: addons/sourcemod/scripting/shavit-replay.sp [hunk 2]
```diff
@@ -1588,17 +1606,40 @@ public void DeleteFrames(int client)
...
+
 public Action Shavit_OnStart(int client)
 {	
 	gI_PlayerPrerunFrames[client] = gA_PlayerFrames[client].Length;
```
Comment (kidfearless) [inline addons/sourcemod/scripting/shavit-replay.sp:1613]: add
```static int s_iLoopFrame[MAXPLAYERS+1];```

File: addons/sourcemod/scripting/shavit-replay.sp [hunk 3]
```diff
@@ -2698,6 +2739,85 @@ void GetReplayName(int style, int track, char[] buffer, int length)
 	Shavit_GetWRName(style, buffer, length, track);
 }
 
+int GetClosetFrameForPlayer(int client, int style, int track)
+{
+	if(!gA_Frames[style][track] || gA_Frames[style][track].Length == 0)
```
Comment (kidfearless) [inline addons/sourcemod/scripting/shavit-replay.sp:2744]: use == null

File: addons/sourcemod/scripting/shavit-replay.sp [hunk 4]
```diff
@@ -2698,6 +2739,85 @@ void GetReplayName(int style, int track, char[] buffer, int length)
...
+	{
+		return -1;
+	}
+
+	float vecPlayerOrigin[3], vecFrameOrigin[3];
+	GetClientAbsOrigin(client, vecPlayerOrigin);
+
+	float flLastDistance = 500.1;
```
Comment (kidfearless) [inline addons/sourcemod/scripting/shavit-replay.sp:2752]: why is this .1?

File: addons/sourcemod/scripting/shavit-replay.sp [hunk 5]
```diff
@@ -1588,17 +1606,40 @@ public void DeleteFrames(int client)
...
+
 public Action Shavit_OnStart(int client)
 {	
 	gI_PlayerPrerunFrames[client] = gA_PlayerFrames[client].Length;
 
+	if(gA_Frames[Shavit_GetBhopStyle(client)][Shavit_GetClientTrack(client)].Length > RoundToFloor(gF_Tickrate * 60))
+	{
+		g_iLoopFrame[client] = gA_Frames[Shavit_GetBhopStyle(client)][Shavit_GetClientTrack(client)].Length / 25;
+	}
+	else
```
Comment (kidfearless) [inline addons/sourcemod/scripting/shavit-replay.sp:1619]: remember to add a new line before else statements

File: addons/sourcemod/scripting/include/shavit.inc [hunk 6]
```diff
@@ -1740,6 +1741,19 @@ native int Shavit_GetPlayerPreFrame(int client);
...
+ * Get preframe length from specific replay
+ * @param stlye						Style
+ * @param track						Track 
+ *
+ * @return Replay preframe 
+ */
+native int Shavit_GetReplayPreFrameCount(int style, int track);
+
+// I need a native English speaker to polish these two natives >_>  ---- deadwinter
+native int Shavit_GetClientCurrentFrame(int client, int style, int track);
```
Comment (kidfearless) [inline addons/sourcemod/scripting/include/shavit.inc:1754]: at least add the comment blocks

File: addons/sourcemod/scripting/shavit-replay.sp (hunk 4 above)
Comment (Nairdaa) [reply addons/sourcemod/scripting/shavit-replay.sp:2752]: because Kamay fucking says so

**Pull request merged**